CefSharp.Wpf/ChromiumWebBrowser.cs

```diff
@@ -1859,37 +1954,15 @@ protected virtual IntPtr SourceHook(IntPtr hWnd, int message, IntPtr wParam, Int
                 return IntPtr.Zero;
             }
 
-            switch ((WM)message)
+            if (!IsDisposed && _imeWin != null && IsKeyboardFocused && browser != null)
```

Comment:
You can now implement your own `IWpfKeyboardHandler` to capture and forward key events.